app/models/grda_warehouse/tasks/push_clients_to_cas.rb

```diff
@@ -74,8 +74,7 @@ def sync!
             GrdaWarehouse::Hud::Project.find_each do |project|
               safe_project_names[project.id] = project.safe_project_name
             end
-            TodoOrDie('test behavior after rails upgrade', if: Rails.version !~ /\A7\.0/)
-            client_source.lazy_preload(preloads).
```

Comment:
not sure if the preloader is still behaving; could use a test to check for n+1

Reply:
@eanders can you point me to how I can exercise this? Maybe I should add a unit test for PushClientsToCas

Reply:
This class requires the CAS database be setup.  We added `lazy_preload` here: https://github.com/greenriver/hmis-warehouse/pull/5236 because it handles some scenarios that we were unable to hit with a standard preload.  

The complexity of the calculations for what is pushed to CAS is installation and configuration specific, often relying on data not in the standard HMIS shapes and specific values of fields to determine what is valid.

I'd love some unit tests, but I think we'll need to add a CAS database (and keep the shape in-sync) to CI before that would be fully possible.  

This precursors to this model generally gets exercised as part of the calculator tests: `spec/models/grda_warehouse/cas_project_client_calculator`.

Reply:
Was there a reason we don't like `lazy_preload`?

Reply:
Thanks, I removed it since we had a todo-or-die to check if the preload behavior was fixed in newer version of rails, in which case the dependency wouldn't be needed.

I was sort of hoping I could add an isolated test case to prove if we needed it or not but maybe that's too ambitious; happy to leave the lazy_preload in for now and we can plan for a future round of dependency pruning.

Reply:
I guess I'm inclined to leave it and then run some manual tests with and without it after 7.1 goes live.  
If it isn't needed we could remove it.  
That said, I think it might be a better overall approach than our sometimes crazy attempts at guessing at preload needs.

Reply:
fair, let's leave it in